### PR TITLE
Pre-select the show all vars checkbox using Institute settings

### LINKED ISSUE #3541: Make "Show variants present only in unaffected" checked by default
Dear developers,

We continue using Scout in teaching and I have rolled forward to the latest code. To not break our lab, I need to have "Show variants present only in unaffected" checked by default in the "Clinical SNV and INDELs" page.

This page is controlled in `scout/server/blueprints/variants/templates/variants/utils.html` but I do not know if there is a way update the html to pass a setting to automatically have it checked when the page loads.

```
{% macro filter_form_footer(form, result_size, total_variants, more_variants, page, nvars) %}
<div class="card-footer text-center">
  <input type="hidden" id="expand_search" name="expand_search">
  <div class="row">
    <div class="col-4">
      <div class="form-check d-flex justify-content-start">
        {{ form.hide_dismissed(class="form-check-input") }}
        {{ form.hide_dismissed.label(class="form-check-label ms-2") }}
      </div>
      <div class="form-check d-flex justify-content-start">
        {{ form.show_unaffected(class="form-check-input") }}
        {{ form.show_unaffected.label(class="form-check-label ms-2", **{'data-bs-toggle': 'tooltip', 'title':'This filter applies only to cases with more than one sample. When this is unchecked, only variants that are predicted by callers to be present in an affected individual genotype are shown.'}) }}
      </div>
    </div>
    <div class="col-4 d-flex justify-content-center">
      Filter returns {{result_size}} / {{ total_variants }} variants.
    </div>
    <div class="col-4 d-flex justify-content-end">
      Showing {%if more_variants %}page {{page}}{%else%}last page{% endif %} with {{nvars}} variants.
    </div>
  </div>
</div>
{% endmacro %}
```

Can someone please help?

image

### PR DESCRIPTION
**Pull request by northwestwitch**
Fix #3541 

- Improve the looks of Institute Settings page
- Adds a new settings option in institutes settings that will force variants filter form to return all variants (not only those present in affected individuals

image

This will cause the variants filter to return all variants:

image

<details>
<summary>Testing on cg-vm1 server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. Make sure the PR is pushed and available on [Docker Hub](https://hub.docker.com/repository/docker/clinicalgenomics/scout-server-stage)
1. Fist book your testing time using the paxa software installed on `hasta`:
 - Log in into hasta: `ssh <USER.NAME>@hasta.scilifelab.se`
 - Activate the staging environment with the command `us` (Use Stage)
 - Run the command `paxa` and follow the instructions. Make sure to specify scout-stage as the resource you request and the server cg-vm1 as server.
1. Log out from the hasta server.
1. `ssh <USER.NAME>@cg-vm1.scilifelab.se`
1. `sudo -iu hiseq.clinical`
1. `ssh localhost`
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `podman ps`
1. Stop the service with current deployed branch: `systemctl --user stop scout.target`
1. Start the scout service with the branch to test: `systemctl --user start scout@<this_branch>`
1. Make sure the branch is deployed: `systemctl --user status scout.target`
1. After testing is done, log out from `cg-vm1` and log in again in the `hasta` server, repeat the `hasta` and `paxa` procedure, which will release the allocated resource (scout-stage) to be used for testing by other users.
</details>

**How to test**:
1. Install on cg-vm1 
2. Edit the settings of one institute to pre-select the `Preselected 'Show also variants only present in unaffected'` checkbox
3. Check variants page (from variants of different type) to make sure that all vars are returned and not only those from affected individuals

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [x] code approved by DN
- [x] tests executed by CR, DN

### CODE REVIEW

File: scout/server/blueprints/institutes/templates/overview/utils.html [hunk 1]
```diff
@@ -1,131 +1,142 @@
 {% macro institute_settings(form, beacon_form, institute, current_user, loqus_instances) %}
-<div class="container">
-
```
Comment (northwestwitch) [inline scout/server/blueprints/institutes/templates/overview/utils.html:3]: These might look a lot of changes, but I've just re-arranged and improved the looks of the page, from this:

image


To this:

image

File: scout/adapter/mongo/event.py [hunk 2]
```diff
@@ -164,11 +164,6 @@ def events(
         if variant_id:
             if comments:
                 # If it's comment-related event collect global and variant-specific comment events
-                LOG.debug(
```
Comment (dnil) [inline scout/adapter/mongo/event.py:167]: 👌

File: scout/server/blueprints/institutes/forms.py [hunk 3]
```diff
@@ -79,12 +79,17 @@ class InstituteForm(FlaskForm):
...
     )
+
     institutes = NonValidatingSelectMultipleField("Institutes to share cases with", choices=[])
 
     loqusdb_id = NonValidatingSelectField("LoqusDB id", choices=[])
 
     alamut_key = StringField("Alamut API key", validators=[validators.Optional()])
 
+    check_show_all_vars = BooleanField(
+        "Preselected 'Show also variants only present in unaffected'"
```
Comment (dnil) [inline scout/server/blueprints/institutes/forms.py:90]:
```suggestion
        "Preselect 'Show also variants only present in unaffected'"
```
perhaps?

File: scout/server/blueprints/institutes/templates/overview/utils.html [hunk 4]
```diff
@@ -1,131 +1,142 @@
...
-              <!-- End of loqusdb settings -->
-              <!-- Alamut settings -->
-              <div class="col-md-3">
-                {{ form.alamut_key.label(class="control-label",data_bs_toggle="tooltip", data_bs_placement="top", title="Optional Alamut Visual Plus API key to be used when opening links to Alamut") }}
-                <a class="ms-3" href="https://extranet.interactive-biosoftware.com/Alamut_Visual_Plus_API_2.0.0.html" target="_blank" rel="noopener">?</a>
-                {{ form.alamut_key(class='form-control', value=institute.alamut_key if institute.alamut_key) }}
+
+              <div class="col-4">
+                {{ form.pheno_groups.label(class="control-label",data_bs_toggle="tooltip", data_bs_placement="top", title="Phenotype groups are used to quickly assign a certain phenotype to a case, on the case page.") }}
+                <select multiple="multiple" name="pheno_groups" class="selectpicker">
```
Comment (dnil) [inline scout/server/blueprints/institutes/templates/overview/utils.html:72]: Same here! 
```suggestion
                <select multiple="multiple" name="pheno_groups" class="selectpicker" data-style="btn-secondary">
```

File: scout/server/blueprints/institutes/templates/overview/utils.html [hunk 5]
```diff
@@ -1,131 +1,142 @@
...
-                {{ form.loqusdb_id.label(class="control-label", data_bs_toggle="tooltip", data_bs_placement="top", title="LoqudDB id" ) }}<br>
-                <select multiple class="selectpicker" name="loqusdb_id" data-style="btn-secondary">
-                  {% for instance in loqus_instances %}
-                    <option value="{{instance}}" {% if instance in institute.loqusdb_id %} selected {% endif %}>{{instance}}</option>
+
+            <div class="row mt-3 d-flex align-items-center">
+
+              <div class="col-4">
+                {{ form.gene_panels.label(class="control-label",data_bs_toggle="tooltip", data_bs_placement="top", title="Select gene panels that will be available for variants filtering.") }}
+                <select multiple="multiple" name="gene_panels" class="selectpicker ml-3">
```
Comment (dnil) [inline scout/server/blueprints/institutes/templates/overview/utils.html:63]: Ahem, *cough, cough* that style for dark mode.. 😉 
```suggestion
                <select multiple="multiple" name="gene_panels" class="selectpicker ml-3" data-style="btn-secondary">
```

**Review by dnil — CHANGES_REQUESTED** [review]
Hi, I might be too early on this one, but it changed from draft! 😊 
It doesn't quite seem to do the job just yet: atm my test have the institute option blocking the checkbox, but blocking it in an unchecked state from the perspective of filtering. I e only variants from affecteds are shown. This could be convenient for the clinic, but we have unchecked as a default so all is well without. The intention for the student project is to be able to block it in a state where also variants from unaffecteds are shown!

**Comment by northwestwitch** [comment]
> Hi, I might be too early on this one, but it changed from draft! 😊
> It doesn't quite seem to do the job just yet: atm my test have the institute option blocking the checkbox, but blocking it in an unchecked state from the perspective of filtering. I e only variants from affecteds are shown. This could be convenient for the clinic, but we have unchecked as a default so all is well without. The intention for the student project is to be able to block it in a state where also variants from unaffecteds are shown!

Ah no, this is not intended, it should be the other way around --> block on checked. I'll fix.

**Comment by northwestwitch** [comment]
> > Hi, I might be too early on this one, but it changed from draft! 😊
> > It doesn't quite seem to do the job just yet: atm my test have the institute option blocking the checkbox, but blocking it in an unchecked state from the perspective of filtering. I e only variants from affecteds are shown. This could be convenient for the clinic, but we have unchecked as a default so all is well without. The intention for the student project is to be able to block it in a state where also variants from unaffecteds are shown!
> 
> Ah no, this is not intended, it should be the other way around --> block on checked. I'll fix.

On which browser? I'm asking because of one recent change I made and I'm not sure it's the same on all browsers.

File: scout/adapter/mongo/institute.py [hunk 6]
```diff
@@ -163,7 +163,7 @@ def update_institute(
                 alamut_key if alamut_key != "" else None
             )  # allows to reset Alamut key to None
 
-        updates["$set"]["check_show_all_vars"] = check_show_all_vars == "on"
```
Comment (northwestwitch) [inline scout/adapter/mongo/institute.py]: Perhaps it was this @dnil, but I'm not sure. This way is safer anyway!

**Comment by northwestwitch** [comment]
I've deployed the latest changes of this branch on stage, for testing!

File: scout/adapter/mongo/institute.py (hunk 6 above)
Comment (dnil) [reply scout/adapter/mongo/institute.py]: Yeah, should be better, but still not working on my local. It's probably around there though.. 😬 
Here's a test (https://scout-stage.scilifelab.se/cust002/F0032570/str/variants):

Screenshot 2022-08-24 at 09 38 36

Screenshot 2022-08-24 at 09 38 45

Screenshot 2022-08-24 at 09 39 47

The goal is to have the two last ones show the same result.. 😸

**Comment by dnil** [comment]
> > > Hi, I might be too early on this one, but it changed from draft! 😊
> > > It doesn't quite seem to do the job just yet: atm my test have the institute option blocking the checkbox, but blocking it in an unchecked state from the perspective of filtering. I e only variants from affecteds are shown. This could be convenient for the clinic, but we have unchecked as a default so all is well without. The intention for the student project is to be able to block it in a state where also variants from unaffecteds are shown!
> > 
> > 
> > Ah no, this is not intended, it should be the other way around --> block on checked. I'll fix.
> 
> On which browser? I'm asking because of one recent change I made and I'm not sure it's the same on all browsers.

Tested on Safari and Chrome. Intuition says a `true` vs `"True"` value type of issue, but I haven't sat down and printed stuff..

File: scout/adapter/mongo/institute.py (hunk 6 above)
Comment (northwestwitch) [reply scout/adapter/mongo/institute.py]: > The goal is to have the two last ones show the same result..

Agreed. I'll check what's wrong!

Comment (northwestwitch) [reply scout/adapter/mongo/institute.py]: STRs page works as expected now:

image

File: scout/server/blueprints/variants/controllers.py [hunk 7]
```diff
@@ -41,6 +41,18 @@
 LOG = logging.getLogger(__name__)
 
 
+def populate_force_show_unaffected_vars(institute_obj, form):
```
Comment (dnil) [inline scout/server/blueprints/variants/controllers.py:44]: 👍

File: scout/server/blueprints/variants/templates/variants/utils.html [hunk 8]
```diff
@@ -67,8 +67,13 @@
         {{ form.hide_dismissed.label(class="form-check-label ms-2") }}
       </div>
       <div class="form-check d-flex justify-content-start">
-        {{ form.show_unaffected(class="form-check-input") }}
-        {{ form.show_unaffected.label(class="form-check-label ms-2", **{'data-bs-toggle': 'tooltip', 'title':'This filter applies only to cases with more than one sample. When this is unchecked, only variants that are predicted by callers to be present in an affected individual genotype are shown.'}) }}
+        {% if institute.check_show_all_vars %}
+          <input type="checkbox" class="form-check-input" name="show_unaffected" id="show_unaffected" checked disabled>
```
Comment (dnil) [inline scout/server/blueprints/variants/templates/variants/utils.html:71]: 👍

**Review by dnil — APPROVED** [review]
Nice and clean!

**Pull request merged**